### ISSUE TITLE
Demo: adding visual embeddings to LayoutLM

### ISSUE THREAD

**Issue by NielsRogge** [microsoft/unilm#314]
A much requested feature/question in this repo was "how do you add visual embeddings to LayoutLM?". I wondered how this worked myself, so (just in time for the release of LayoutLM 2.0), here's a notebook that fine-tunes LayoutLM on the [FUNSD](https://guillaumejaume.github.io/FUNSD/) dataset, thereby adding visual embeddings from a pre-trained ResNet-101 backbone:

https://github.com/NielsRogge/Transformers-Tutorials/blob/master/LayoutLM/Add_image_embeddings_to_LayoutLM.ipynb

It uses ROI-align to turn each bounding box into a feature map, which is then projected to match the hidden_size of LayoutLM (which is 768 for the base model). By adding these visual features, I was able to improve performance on the test set compared to using only text + layout (bounding boxes) information to around the following:

'precision': 0.8053668087066682, 'recall': 0.8163670324538874, 'f1': 0.8108296133109165

Related issues:

#201 
#95 
#265 
#80 
#243 
#286
#285 
#249 
#165 
#97

**Comment by victor-ab** [comment]
@NielsRogge Nice work, thanks for sharing!
What was the f1 score without the image features?

**Comment by NielsRogge** [comment]
Hi, thank you!

As can be seen in my [earlier notebook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/LayoutLM/Fine_tuning_LayoutLMForTokenClassification_on_FUNSD.ipynb), the performance was:

'precision': 0.7239292364990689, 'recall': 0.7778889444722361, 'f1': 0.7499397154569568

**Comment by ruifcruz** [comment]
You rock!! 🚀

**Comment by fredo838** [comment]
Respect!

**Comment by brunnurs** [comment]
Your work for the NLP/NLU community, especially for us guys trying to apply this papers to use cases, is extremely helpful! Many thanks and keep up the good work.

**Comment by hamavb** [comment]
is this the exact algo that used to produce the SOA results by the team ?

nice work by the way, thxs !!
